cli/azd/pkg/azure/arm_parameters.go

```diff
@@ -0,0 +1,16 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package azure
+
+type ArmParameters map[string]ArmParameterValue
```

Comment:
I was getting lost between `ArmParameters` and `ArmTemplateParameters`. I see now that the distinction is indeed in the name, and I was just being careless. Naming something like `ArmParameterFileParameters` might be too much, so maybe a middleground is to simply mention in comments `// The parameters array from deployment the parameters file (typically named *.parameters.json)`

Reply:
Hmm I see that my initial assessment is inaccurate as we would use the same type to mean the user-provided parameters.

Is it simply that `ArmTemplateParameters` needs to be renamed to `ArmParameterDefinitions`? or `ArmTemplateParametersDefinition`?

From the schema [docs](https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json):

```
parameters: {
type: "object",
description: "Input parameter definitions",
additionalProperties: {
$ref: "#/definitions/parameter"
}
},
```

Reply:
Thank you! This is a much better name, and I was really struggling here.